controllers/database_reconciler_helpers.go

```diff
@@ -253,3 +258,124 @@ func (r *DatabaseReconciler) handleSync(ctx context.Context, database *ndbv1alph
+	log := ctrllog.FromContext(ctx)
+	log.Info("Entered database_reconciler_helpers.setupConnectivity")
+	// The 'service' and 'endpoint' objects should have the
+	// same name for the service to map to the enpoint.
+	commonMetadata := metav1.ObjectMeta{
+		Name:      database.Name + "-svc",
+		Namespace: req.Namespace,
+	}
+	commonNamespacedName := types.NamespacedName{
+		Name:      database.Name + "-svc",
+		Namespace: req.Namespace,
+	}
+	targetPort := ndbv1alpha1.GetDatabasePortByType(database.Spec.Instance.Type)
+
+	err = r.setupService(ctx, database, commonNamespacedName, commonMetadata, targetPort)
+	if err != nil {
+		log.Error(err, "Error occurred while setting up the service")
+		return
+	}
+	err = r.setupEndpoints(ctx, database, commonNamespacedName, commonMetadata, targetPort)
+	if err != nil {
+		log.Error(err, "Error occurred while setting up the endpoints")
+		return
+	}
+	log.Info("Returning from database_reconciler_helpers.setupConnectivity")
+	return
+}
+
+// Checks and creates a new service (without label selectors) if it does not exists
+// and also sets up the database as the owner for the created service
+func (r *DatabaseReconciler) setupService(ctx context.Context, database *ndbv1alpha1.Database, namespacedName types.NamespacedName, metadata metav1.ObjectMeta, targetPort int32) (err error) {
+	log := ctrllog.FromContext(ctx)
+	log.Info("Entered database_reconciler_helpers.setupService")
+	// Create a new service if it does not exists.
+	foundService := &corev1.Service{}
+	err = r.Get(ctx, namespacedName, foundService)
+	if err != nil && errors.IsNotFound(err) {
+		log.Info("No service found, creating a new service", "target port", targetPort)
+		service := &corev1.Service{
+			ObjectMeta: metadata,
+			Spec: corev1.ServiceSpec{
+				Ports: []corev1.ServicePort{
+					{
+						Protocol:   corev1.ProtocolTCP,
+						Port:       80,
+						TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: targetPort},
+					},
+				},
+			},
+		}
+		// Setting database as the owner of this service
+		err = ctrl.SetControllerReference(database, service, r.Scheme)
+		if err != nil {
+			log.Error(err, "Error setting controller reference for the service")
+		}
+		err = r.Create(ctx, service)
+		if err != nil {
+			log.Error(err, "Failed to create a new service")
+			return
+		}
+		log.Info("Created a new service", "service name", service.GetName())
+	}
+	log.Info("Returning from database_reconciler_helpers.setupService")
+	return
+}
+
+// Checks and creates an endpoints object for the service if it does not already exists.
+// If it is already present, syncs the IP address with the Database.Status.IPAddress if out of sync.
+func (r *DatabaseReconciler) setupEndpoints(ctx context.Context, database *ndbv1alpha1.Database, namespacedName types.NamespacedName, metadata metav1.ObjectMeta, targetPort int32) (err error) {
+	log := ctrllog.FromContext(ctx)
+	log.Info("Entered database_reconciler_helpers.setupEndpoints")
```

Comment:
Rather than this type of comments, would it make sense to say that you are starting to set endpoints for a given database, under a given namespace, etc

Reply:
Not really, this method will be executed on every reconcile request (even after the endpoints have been set in an earlier reconcile). In that case, this method won't create / set endpoint. However, it will check if the endpoints have been changed and log an appropriate message later.